specs/IsSmartScreenRequired.md

```diff
@@ -0,0 +1,99 @@
+# Background
+
+[Edge SmartScreen](https://support.microsoft.com/en-us/microsoft-edge/how-can-smartscreen-help-protect-me-in-microsoft-edge-1c9a874a-6826-be5e-45b1-67fa445a74c8) helps end users identify reported phishing and malware websites, and also helps end users make informed decisions about downloads.
+
+Currently, developers can use `options->put_AdditionalBrowserArguments(L"--disable-features=msSmartScreenProtection")` to disable SmartScreen in the WebView2 application. It is essentially a startup parameter of the browser process and applies to all WebView2 instances associated with that WebView2Environment. It must be determined when the WebView2Environment is created, and it cannot be modified at runtime.
+
+To support more flexibility we introduce a new API.
+
+we have CoreWebView2Settings.IsSmartScreenRequired. Each WebView2 declares if it requires SmartScreen. Some WebView2s may be used to display app content and don't require SmartScreen and others may be rendering arbitrary web content and do need SmartScreen. Having SmartScreen on unnecessarily for app content is a detriment to performance but otherwise not a problem. Having SmartScreen off for arbitrary web content is an issue. We have to turn SmartScreen on or off for all the WebView2s in an environment so if any WebView2 requires SmartScreen then we turn it on for all of them. If WebView2 settings change or WebView2s are closed and then all the WebView2s in an environment don't require SmartScreen, then we can turn SmartScreen off.
+
+It is much easier to indicate if individual WebView2s require SmartScreen than to have an end developer explicitly manage if SmartScreen should be enabled as a whole, especially when its different sets of WebView2s in different processes (like Excel's set of WebView2s and Word's set of WebView2s) all sharing the same user data folder.
+In this document we describe the new setting.
+
+
+# Description
+You can use CoreWebView2Settings.IsSmartScreenRequired to control SmartScreen. SmartScreen is enabled or disabled per browser process, so all WebView2 applications sharing the same user data folder path also share SmartScreen being enabled or disabled.
```

Comment:
Not quite.

Later the documentation is in terms of CoreWebView2Environment and it may be clearer to say that instead of browser process and user data folder throughout the document (as appropriate).

SmartScreen is enabled or disabled for all the CoreWebView2s in a CoreWebView2Environment. Each CoreWebView2 can set the IsSmartScreenRequired saying whether that CoreWebView2 in particular requires SmartScreen. If any CoreWebView2 in the CoreWebView2Environment requires SmartScreen, then its enabled. If none do, then it is disabled. This works because turning on SmartScreen unnecessarily doesn't cause functional problems, just a waste of perf. So if all the CoreWebView2s in a CoreWebView2Environment agree to opt-out of SmartScreen, then we can turn it off and save perf. If any CoreWebView2 requires it, then its enabled.